### PR TITLE
[Automation] Migrate and enable github happy path flow test case

### PR DESCRIPTION
**Pull request by subhashkhileri**
## Description

[Automation] Migrate and enable github happy path flow test case - [RHIDP-1109](https://issues.redhat.com//browse/RHIDP-1109)

## Which issue(s) does this PR fix

https://issues.redhat.com/browse/RHIDP-1109

## PR acceptance criteria

Please make sure that the following steps are complete:

- [ ] GitHub Actions are completed and successful
- [ ] Unit Tests are updated and passing
- [ ] E2E Tests are updated and passing
- [ ] Documentation is updated if necessary (requirement for new features)
- [ ] Add a screenshot if the change is UX/UI related

### CODE REVIEW

File: e2e-tests/playwright/e2e/github-happy-path.spec.ts [hunk 1]
```diff
@@ -16,18 +16,30 @@ test.describe.skip('GitHub Happy path', () => {
   const component =
     'https://github.com/janus-idp/backstage-showcase/blob/main/catalog-entities/all.yaml';
 
-  test.beforeAll(async ({ browser }) => {
+  const resources = [
+    'Janus-IDP Authors',
```
Comment (gustavolira) [inline e2e-tests/playwright/e2e/github-happy-path.spec.ts:20]: Can we use it externally from the test? Like on Page object page, so we can reuse it in another tests if needed

File: e2e-tests/playwright/e2e/github-happy-path.spec.ts [hunk 2]
```diff
@@ -89,14 +100,80 @@ test.describe.skip('GitHub Happy path', () => {
     await expect(page.locator(`text=${issuesCountText}`)).toBeVisible();
 
     for (const issue of openIssues.slice(0, 5)) {
-      const issueLocator = page.locator(
-        `text=${issue.title.replace(/\s+/g, ' ')}`,
+      const issueSelector = `text=${issue.title.replace(/\s+/g, ' ')}`;
```
Comment (gustavolira) [inline e2e-tests/playwright/e2e/github-happy-path.spec.ts:103]: can you create `(/\s+/g` as a constant and give a name which explains the usage?

File: e2e-tests/playwright/e2e/github-happy-path.spec.ts [hunk 3]
```diff
@@ -89,14 +100,80 @@ test.describe.skip('GitHub Happy path', () => {
...
+    const openPRs = await BackstageShowcase.getGithubPRs('open');
+    backstageShowcase.verifyPRRows(openPRs, 0, 5);
+  });
+
+  test('Click on the CLOSED filter and verify that the 5 most recently updated Closed PRs are rendered (same with ALL)', async () => {
+    await uiHelper.clickButton('CLOSED', { force: true });
+    const closedPRs = await BackstageShowcase.getGithubPRs('closed');
+
+    for (const closedPR of closedPRs.slice(0, 5)) {
+      await expect(page.locator(`text=${closedPR.title}`)).toBeVisible();
```
Comment (gustavolira) [inline e2e-tests/playwright/e2e/github-happy-path.spec.ts:121]: can we use `page.locator(`text=${closedPR.title}`)` on UiHelper?

File: e2e-tests/playwright/e2e/github-happy-path.spec.ts [hunk 4]
```diff
@@ -89,14 +100,80 @@ test.describe.skip('GitHub Happy path', () => {
...
+  test('Click on the arrows to verify that the next/previous/first/last pages of PRs are loaded', async () => {
+    const allPRs = await BackstageShowcase.getGithubPRs('all', true);
+
+    await uiHelper.clickButton('ALL', { force: true });
+    await backstageShowcase.verifyPRRows(allPRs, 0, 5);
+
+    await backstageShowcase.clickNextPage();
+    await backstageShowcase.verifyPRRows(allPRs, 5, 10);
+
+    const lastPagePRs = Math.floor((allPRs.length - 1) / 5) * 5;
```
Comment (gustavolira) [inline e2e-tests/playwright/e2e/github-happy-path.spec.ts:134]: can it be extract to a method with a documentation explaining what the method is doing, this way we avoid magic numbers

File: e2e-tests/playwright/e2e/github-happy-path.spec.ts [hunk 5]
```diff
@@ -89,14 +100,80 @@ test.describe.skip('GitHub Happy path', () => {
...
+  });
+
+  test('Verify that the CI tab renders 5 most recent github actions and verify the table properly displays the actions when page sizes are changed and filters are applied', async () => {
+    await uiHelper.clickTab('CI');
+    await common.clickOnGHloginPopup();
+
+    const workflowRuns = await backstageShowcase.getWorkflowRuns();
+
+    for (const workflowRun of workflowRuns.slice(0, 5)) {
+      await expect(page.locator(`text=${workflowRun.id}`)).toBeVisible();
```
Comment (gustavolira) [inline e2e-tests/playwright/e2e/github-happy-path.spec.ts:157]: can we use UiHelper for `page.locator(`text=${workflowRun.id}`)` ?

File: e2e-tests/playwright/support/pages/CatalogImport.ts [hunk 6]
```diff
@@ -56,47 +54,47 @@ export class BackstageShowcase {
...
-    if (numberExists) {
-      throw new Error(
-        `Element with number ${allPRs[rows].number} should not exist`,
-      );
-    }
+    const rowElement = this.page.locator(`text=${allPRs[rows - 1].title}`);
+    await rowElement.scrollIntoViewIfNeeded();
+    await expect(rowElement).toBeVisible();
+
+    const nonExistentElement = this.page.locator(
```
Comment (gustavolira) [inline e2e-tests/playwright/support/pages/CatalogImport.ts:70]: can we use UiHelper here? for this page.locator?

File: e2e-tests/playwright/support/pages/CatalogImport.ts [hunk 7]
```diff
@@ -109,4 +107,12 @@ export class BackstageShowcase {
       throw new Error('About card is not displayed');
     }
   }
+
+  async verifyPRRows(allPRs: any[], startRow: number, lastRow: number) {
+    for (let i = startRow; i < lastRow; i++) {
+      await expect(
+        this.page.locator(`td:has-text("${allPRs[i].title}")`).first(),
```
Comment (gustavolira) [inline e2e-tests/playwright/support/pages/CatalogImport.ts:114]: can we use UiHelper here?

File: e2e-tests/playwright/support/pages/CatalogImport.ts [hunk 8]
```diff
@@ -56,47 +54,47 @@ export class BackstageShowcase {
...
   async verifyPRStatisticsRendered() {
     const regex = /Average Size Of PR\d+ lines/;
-    const statsVisible = await this.page
-      .locator('tr')
-      .filter({ hasText: regex })
-      .isVisible();
-    if (!statsVisible) {
-      throw new Error('PR statistics are not rendered');
-    }
+    const stats = this.page.locator('tr').filter({ hasText: regex });
```
Comment (gustavolira) [inline e2e-tests/playwright/support/pages/CatalogImport.ts:95]: I think you can use uiHelper for all those generic situations, and move `waitFor` to the method inside UIHelper

File: e2e-tests/playwright/e2e/github-happy-path.spec.ts (hunk 2 above)
Comment (subhashkhileri) [reply e2e-tests/playwright/e2e/github-happy-path.spec.ts:103]: It's a simple, self-explanatory regex. It is replacing two or more spaces with a single space.

**Pull request merged**